### PR TITLE
Slither findings

### PR DESCRIPTION
**Pull request by grandizzy**

### CODE REVIEW

**Review by EdNoepel — APPROVED** [review]
LGTM

File: src/base/FenwickTree.sol [hunk 1]
```diff
@@ -35,7 +35,7 @@ abstract contract FenwickTree {
         require(f_ != 0,   "FW:M:FACTOR_ZERO");
 
         i_ += 1;
-        uint256 sum;
+        uint256 sum = 0;
```
Comment (ith-harvey) [inline src/base/FenwickTree.sol:38]: why add this?

File: src/base/FenwickTree.sol [hunk 2]
```diff
@@ -46,13 +46,17 @@ abstract contract FenwickTree {
...
             sum = scaledI != 0 ? sum + Maths.wmul(Maths.wmul(df, values[i_]), scaledI) : sum + Maths.wmul(df, values[i_]);
             scaling[i_] = scaledI != 0 ? Maths.wmul(f_, scaledI) : f_;
-            j = i_ + _lsb(i_);
-            i_ -= _lsb(i_);
-            while ((_lsb(j) < _lsb(i_)) || (i_ == 0 && j <= SIZE)) {
+            uint256 lsbI = _lsb(i_);
+            j = i_ + lsbI;
+            i_ -= lsbI;
+            uint256 lsbJ = _lsb(j);
+            //slither-disable-next-line incorrect-equality
```
Comment (ith-harvey) [inline src/base/FenwickTree.sol:53]: Should this comment still exist in the code?

File: src/base/FenwickTree.sol [hunk 3]
```diff
@@ -61,21 +65,23 @@ abstract contract FenwickTree {
         require(i_ < SIZE, "FW:A:INVALID_INDEX");
 
         i_ += 1;
-        uint256 j = 8192; // 1 << 13
-        uint256 ii;
-        uint256 sc = Maths.WAD;
+        uint256 j     = 8192; // 1 << 13
+        uint256 ii    = 0;
```
Comment (ith-harvey) [inline src/base/FenwickTree.sol:69]: Why set these to zero?

**Review by ith-harvey — COMMENTED** [review]
A couple questions... Also would you mind outlining what you changed in your PR description? a bullet or two would be helpful

**Comment by grandizzy** [comment]
> 

each commit address a specific fix explained in commit comment and when merging with squash they'll nicely show up together, I recommend checking them one by one if that's OK

File: src/base/FenwickTree.sol (hunk 2 above)
Comment (grandizzy) [reply src/base/FenwickTree.sol:53]: yeah, this is to disable following warning:

```
FenwickTree._mult(uint256,uint256) (src/base/FenwickTree.sol#33-61) uses a dangerous strict equality:
	- (lsbJ < _lsb(i_)) || (i_ == 0 && j <= SIZE) (src/base/FenwickTree.sol#53)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#dangerous-strict-equalities
```

see https://github.com/ajna-finance/contracts/pull/206/commits/fa0f6c56817c1577a7b7c4c86470f0644df981a8

I didn't want to disable it globally so did it for this specific line

File: src/base/FenwickTree.sol (hunk 3 above)
Comment (grandizzy) [reply src/base/FenwickTree.sol:69]: That's to fix following warnings:

```
FenwickTree Fix FenwickTree._findSum(uint256).ss (src/base/FenwickTree.sol#135) is a local variable never initialized
FenwickTree._add(uint256,uint256).ii (src/base/FenwickTree.sol#65) is a local variable never initialized
FenwickTree._prefixSum(uint256).ii (src/base/FenwickTree.sol#108) is a local variable never initialized
FenwickTree._mult(uint256,uint256).sum (src/base/FenwickTree.sol#38) is a local variable never initialized
FenwickTree._remove(uint256,uint256).ii (src/base/FenwickTree.sol#87) is a local variable never initialized
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#uninitialized-local-variables
```
see commit https://github.com/ajna-finance/contracts/pull/206/commits/3236a90c57a5ed15b58df89540b727e652e3f54a
I compared gas costs and is the same so wanted to make slither happy :)

File: src/base/FenwickTree.sol (hunk 2 above)
Comment (ith-harvey) [reply src/base/FenwickTree.sol:53]: hmm... so we will always have these lines in comments even when we deploy to prod or just while the project is in development?

Comment (ith-harvey) [reply src/base/FenwickTree.sol:53]: Spoke with George and will bring up to rest of the team -> Slither comments will remain in the repo until deployment, then they will be removed for code cleanliness purposes : )

**Review by ith-harvey — APPROVED** [review]

**Review by MikeHathaway — APPROVED** [review]
LGTM

**Pull request merged**